pkg/manager/member/tikv_member_manager.go

```diff
@@ -607,9 +607,10 @@ func getTikVConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
 	scriptModel := &TiKVStartScriptModel{
 		EnableAdvertiseStatusAddr: false,
 		DataDir:                   filepath.Join(tikvDataVolumeMountPath, tc.Spec.TiKV.DataSubDir),
+		ClusterDomain:             tc.Spec.ClusterDomain,
 	}
 	if tc.Spec.EnableDynamicConfiguration != nil && *tc.Spec.EnableDynamicConfiguration {
-		scriptModel.AdvertiseStatusAddr = "${POD_NAME}.${HEADLESS_SERVICE_NAME}.${NAMESPACE}.svc"
+		scriptModel.AdvertiseStatusAddr = "${POD_NAME}.${HEADLESS_SERVICE_NAME}.${NAMESPACE}.svc" + controller.FormatClusterDomain(tc.Spec.ClusterDomain)
```

Comment:
We may have to update the store pattern here https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/tikv_member_manager.go#L52, can you see the stores in the status section of the TidbCluster CR with the current implementation?

Reply:
TiFlash has the same issue.

Reply:
May also have to add a match pattern for PD and TiDB so that the member info can only include those members for this TidbCluster.